src/main/java/uk/gov/hmcts/reform/sscs/service/RetryNotificationService.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import static java.util.Objects.nonNull;
+
+import java.time.ZonedDateTime;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.config.RetryConfig;
+import uk.gov.hmcts.reform.sscs.exception.NotificationServiceException;
+import uk.gov.hmcts.reform.sscs.factory.NotificationWrapper;
+import uk.gov.service.notify.NotificationClientException;
+
+@Service
+@Slf4j
+public class RetryNotificationService {
+    private NotificationHandler notificationHandler;
+    private RetryConfig retryConfig;
+
+    @Autowired
+    RetryNotificationService(NotificationHandler notificationHandler, RetryConfig retryConfig) {
+        this.notificationHandler = notificationHandler;
+        this.retryConfig = retryConfig;
+    }
+
+    public void rescheduleIfHandledGovNotifyErrorStatus(final int retry,
+                                                        final NotificationWrapper notificationWrapper,
+                                                        final NotificationServiceException e) {
+        if (nonNull(e.getCause()) && e.getCause() instanceof NotificationClientException) {
```

Comment:
As we are throwing NotificationClientException so directly we can check this exception, why we are checking 400 && 403 here

Reply:
There are some un-recoverable errors from gov notify, we do not want to re-try these.
https://docs.notifications.service.gov.uk/java.html#error-codes
400 and 403 are determined by the tickets acceptance criteria as being unrecoverable